src/lib/containers/drug_upload_tool/NavButtons.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import { Step, NavActionEnum } from './types';
+import {
+  faChevronLeft,
+  faChevronRight
+} from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+export interface NavButtonsProps {
+  isWizardMode?: boolean;
+  previousStepIds: string[];
+  onNavAction: Function;
+  steps: Step[];
+  currentStep: Step;
+}
+
+export interface NextStepLinkProps {
+  onNavAction: Function;
+  steps: Step[];
+  nextStepId: string | undefined;
+}
+
+export function NavButtons(props: NavButtonsProps): JSX.Element {
+  // in wizard mode we build an array of the previous steps. In regular mode back goes to
+  // a previous order step
+  const canGoBack = (props: NavButtonsProps): boolean => {
+    if (props.isWizardMode) {
+      return props.previousStepIds && props.previousStepIds.length > 0;
+    } else {
+      return (
+        props.steps.findIndex(step => step.id === props.currentStep.id) > 0
+      );
+    }
+  };
+
+  const previousButton = canGoBack(props) ? (
+    <button
+      type="button"
+      onClick={e => props.onNavAction(NavActionEnum.PREVIOUS)}
+      className="btn btn-link nav-link prev"
+    >
+      <FontAwesomeIcon icon={faChevronLeft} />
+    </button>
+  ) : (
+    <></>
+  );
+
+  const nextButton = !props.currentStep.final ? (
+    <button
+      type="button"
+      onClick={e => props.onNavAction(NavActionEnum.NEXT)}
+      className="btn btn-link nav-link next"
+    >
+      <FontAwesomeIcon icon={faChevronRight} />
+    </button>
+  ) : (
+    <></>
+  );
+
+  const saveButton = !props.currentStep.final ? (
+    <button
+      type="button"
+      className="btn btn-action save"
+      onClick={e => props.onNavAction(NavActionEnum.SAVE)}
+    >
+      SAVE
+    </button>
+  ) : (
+    <></>
+  );
+
+  return (
+    <div>
+      <hr></hr>
+      <div className="buttonWrapper pull-right">
+        {previousButton} {nextButton} {saveButton}
+      </div>
+    </div>
+  );
+}
+
+export function NextStepLink(props: NextStepLinkProps): JSX.Element {
+  const nextStep = props.steps.find(step => step.id === props.nextStepId);
+  if (typeof nextStep === 'undefined') {
+    return <></>;
+  }
+  return (
+    <span className="nav-link">
+      <a onClick={e => props.onNavAction(nextStep)}>{nextStep.title}&nbsp;</a>{' '}
```

Comment:
lot of odd spaces here (nbsp and a ' '?).  could you just set margin right instead of these solutions?